libraries/AP_Scripting/examples/quadruped.lua

```diff
@@ -0,0 +1,308 @@
+            (leg_step==-1 or leg_step==(gait_step_total-1)))) and move_requested) then
+        gait_pos_x[moving_leg] = -x_speed/gait_lift_divisor
+        gait_pos_z[moving_leg] = -3*leg_lift_height/(3+gait_half_lift_height)
+        gait_pos_y[moving_leg] = -y_speed/gait_lift_divisor
+        gait_rot_z[moving_leg] = -yaw_speed/gait_lift_divisor
+
+    elseif ((gait_lifted_steps>=2) and (leg_step==1 or leg_step==-(gait_step_total-1)) and move_requested) then
+        gait_pos_x[moving_leg] = x_speed/gait_lift_divisor
+        gait_pos_z[moving_leg] = -3*leg_lift_height/(3+gait_half_lift_height)
+        gait_pos_y[moving_leg] = y_speed/gait_lift_divisor
+        gait_rot_z[moving_leg] = yaw_speed/gait_lift_divisor
+
+    elseif (((gait_lifted_steps==5 and (leg_step==-2 ))) and move_requested) then
+        gait_pos_x[moving_leg] = -x_speed/2
+        gait_pos_z[moving_leg] = -leg_lift_height/2
+        gait_pos_y[moving_leg] = -y_speed/2
+        gait_rot_z[moving_leg] = -yaw_speed/2
+
+    elseif ((gait_lifted_steps==5) and (leg_step==2 or leg_step==-(gait_step_total-2)) and move_requested) then
+        gait_pos_x[moving_leg] = x_speed/2
+        gait_pos_z[moving_leg] = -leg_lift_height/2
+        gait_pos_y[moving_leg] = y_speed/2
+        gait_rot_z[moving_leg] = yaw_speed/2
+
+    elseif ((leg_step==gait_down_steps or leg_step==-(gait_step_total-gait_down_steps)) and gait_pos_y[moving_leg]<0) then
+        gait_pos_x[moving_leg] = x_speed/2
+        gait_pos_z[moving_leg] = y_speed/2
+        gait_pos_y[moving_leg] = yaw_speed/2
+        gait_rot_z[moving_leg] = 0
+
+    else
+        gait_pos_x[moving_leg] = gait_pos_x[moving_leg] - (x_speed/gait_speed_divisor)
+        gait_pos_z[moving_leg] = 0
+        gait_pos_y[moving_leg] = gait_pos_z[moving_leg] - (y_speed/gait_speed_divisor)
+        gait_rot_z[moving_leg] = gait_rot_z[moving_leg] - (yaw_speed/gait_speed_divisor)
+    end
+end
+
+-- Body Forward Kinematics calculates where each leg should be.
+-- inputs are
+--   a) body rotations: body_rot_x, body_rot_y, body_rot_z
+--   b) body position: body_pos_x, body_pos_y, body_pos_z
+--   c) offset of the center of body
+function body_forward_kinematics(X, Y, Z, Xdist, Ydist, Zrot)
+    local totaldist = {X + Xdist + body_pos_x, Y + Ydist + body_pos_y}
+    local distBodyCenterFeet = math.sqrt(totaldist[1]^2 + totaldist[2]^2)
+    local AngleBodyCenter = math.atan(totaldist[2], totaldist[1])
+    local rolly = math.tan(body_rot_y * math.pi/180) * totaldist[1]
+    local pitchy = math.tan(body_rot_x * math.pi/180) * totaldist[2]
+
+    local ansx = math.cos(AngleBodyCenter + ((body_rot_z+Zrot)  * math.pi/180)) * distBodyCenterFeet - totaldist[1] + body_pos_x
+    local ansy = math.sin(AngleBodyCenter + ((body_rot_z+Zrot) * math.pi/180)) * distBodyCenterFeet - totaldist[2] + body_pos_y
+    local ansz = rolly + pitchy + body_pos_z
+    local ans = {ansx, ansy, ansz}
```

Comment:
Why bind to the local instead of just returning? Not sure it actually tweaks the generated bytecode at all, but can be nicer to read. (And keeps the script a bit shorter)

Reply:
thanks, directly return the array